### ISSUE TITLE
webpack errors when using @expo/cli

### ISSUE THREAD

**Issue by leggomuhgreggo** [nrwl/nx#14789]
### Current Behavior

Running `npx expo start --web`, using the new `@expo/cli` package, throws a webpack error: `ValidationError: Invalid options object` related to version mismatch between `webpack@4` and `webpack@5`.

![image](https://user-images.githubusercontent.com/2213636/216538256-e8925231-729b-467a-b76d-b3f5830d19c8.png)

The workaround, for some time, has been to run `npx expo-cli start --web` which uses the legacy `expo-cli` package, which prints a somewhat imposing warning message:
![image](https://user-images.githubusercontent.com/2213636/216534393-bced144b-ec06-41be-9733-f52b9b57ee6b.png)

### Expected Behavior

ability to start expo web apps using standard `@expo/cli` tool.

### GitHub Repo

https://github.com/leggomuhgreggo/nx-issue-expo-babel/pull/2

### Steps to Reproduce

1. clone, checkout `repro/new-cli-not-working` branch, yarn install
2. run app with `yarn nx run repro-app:run-web`
3. observe webpack error

```
git clone git@github.com:leggomuhgreggo/nx-issue-expo-babel.git \
  && cd $_
  && git fetch origin repro/new-cli-not-working
  && git checkout repro/new-cli-not-working
  && yarn install
  && yarn nx run repro-app:run-web
```

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.15.1
   OS   : darwin x64
   yarn : 1.22.19
   
   nx : 15.6.3
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : 15.6.3
   @nrwl/devkit : 15.6.3
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.6.3
   @nrwl/expo : 15.6.3
   @nrwl/express : Not Found
   @nrwl/jest : 15.6.3
   @nrwl/js : 15.6.3
   @nrwl/linter : 15.6.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.6.3
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : 15.6.3
   @nrwl/workspace : 15.6.3
   @nrwl/vite : Not Found
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:

✨  Done in 0.74s.
```

### Failure Logs

```shell
ValidationError: Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration.node should be one of these:
   false | object { __dirname?, __filename?, global? }
   -> Include polyfills or mocks for various node stuff.
   Details:
    * configuration.node has an unknown property 'module'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'dgram'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'dns'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'fs'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'http2'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'net'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'tls'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'child_process'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
```

### Additional Information

Here's a [workaround](https://github.com/leggomuhgreggo/nx-issue-expo-babel/pull/2) which adds a yarn resolution for webpack@4, and removes the legacy expo-cli. 

Some additional context in this [Nx slack thread](https://nrwlcommunity.slack.com/archives/C028DKC9X9A/p1670963038924399)

Thanks!

**Comment by groger** [comment]
same issue here ! +1

**Comment by xiongemi** [comment]
submit a pr https://github.com/nrwl/nx/pull/14944 to fix this

**Comment by xiongemi** [comment]
this should be fixed with nx version 15.7.0
